python/drned_xmnr/op/transitions_op.py

```diff
@@ -182,13 +178,37 @@ def perform_transitions(self):
         return result
 
 
-class WalkTransitionsOp(ExploringOp):
+class WalkTransitionsOp(TransitionsOp):
     action_name = 'walk states'
 
     def _init_params(self, params):
-        super(WalkTransitionsOp, self)._init_params(params)
+        pstates = list(params.states)
+        if pstates == []:
+            pstates = list(self.filter_state_sets(self.get_states()))
+            random.shuffle(pstates)
```

Comment:
I wonder about the randomness, I agree that it's a good idea in general but sometime you probably want to run the test in the same order?

On the other hand, you can of course just specify  the relevant tests explicitly...

Reply:
The order of states is by default somewhat arbitrary (not random) anyway, given by how the filesystem orders the state files; I believe it is better to make it completely random. And yes, as you write, if particular sequence is needed, it can be specified by the user.